src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,304 @@
+    // We don't want the form to actually submit the page, just to enable
+    // activating the submit button on Enter
+    event.preventDefault();
+
+    // Validate context name, eventually this will be replaced with
+    // our new json schema work.
+    if (that.model.contextId === "") {
+        that.applier.change("contextIdErrors", "Please enter an ID.");
+        that.reRender();
+        return;
+    }
+    // TODO standardize the accepted regular expression for prefset ID's
+    // and other parts of the schema.
+    else if (!/^[\w-]+$/.exec(that.model.contextId)) {
+        that.applier.change("contextIdErrors", "Please use only alphanumeric characters.");
+        that.reRender();
+        return;
+    }
+    // If we changed the contextId, make sure that there isn't an existing context with
+    // that ID
+    else if (that.model.contextId !== that.model.originalContextId &&
+             editPrefs.prefsetExists(that.model.contextId)) {
+        that.applier.change("contextIdErrors", "A Preference Set with that ID already exists.");
+        that.reRender();
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        that.applier.change("contextIdErrors", "");
+    }
+
+    if (that.model.contextName === "") {
+        that.applier.change("contextNameErrors", "Please enter a name.");
+        that.reRender();
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        that.applier.change("contextNameErrors", "");
+    }
+
+    var transaction = editPrefs.applier.initiate();
+    var prefsetToEdit = fluid.copy(editPrefs.model.flatPrefs.contexts[that.model.originalContextId]);
+    prefsetToEdit.name = that.model.contextName;
+    var oldPath = "flatPrefs.contexts." + that.model.originalContextId;
+    var path = "flatPrefs.contexts." + that.model.contextId;
+
+    transaction.fireChangeRequest({
+        path: oldPath,
+        type: "DELETE"
+    });
+
+    transaction.fireChangeRequest({
+        path: path,
+        value: prefsetToEdit
+    });
+
+    transaction.commit();
+    that.closeDialog();
+};
+
+
+/**
+ * Confirm deletion of NP Set Context Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmDeleteContextDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-confirmDeleteContext-dialog"
+    },
+    model: {
+        contextId: "" // Should be populated/relayed during construction before showing dialog
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.confirmDeleteContextDialog.acceptConfirmDialog",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{that}.model.contextId"]
+        }
+    }
+});
+
+gpii.devpmt.dialogs.confirmDeleteContextDialog.acceptConfirmDialog = function (that, editPrefs, contextId) {
+    that.closeDialog();
+    var path = "flatPrefs.contexts." + contextId;
```

Comment:
`contextId` probably should be validated before the deletion.

Reply:
Also updated.

Reply:
I don't believe an extra check is required here because of how the changeApplier works, but have added a comment to that extent at least.